### PR TITLE
keep multiple filter

### PR DESCRIPTION
**Pull request by jzhangs**
under multi mode, no need to clear input filter after menu select or deslect

### CODE REVIEW

**Comment by jzhangs** [comment]
original issue in antd [https://github.com/ant-design/ant-design/issues/9573](https://github.com/ant-design/ant-design/issues/9573)

**Comment by yutingzhao1991** [comment]
@jzhangs could you add a test case for this?

**Comment by jzhangs** [comment]
@yutingzhao1991 done.

File: src/Select.jsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {
   getPropValue,
   getValuePropValue,
   isCombobox,
+  isMultiple,
```
Comment (yutingzhao1991) [inline src/Select.jsx]: I think tag should use the same logic. so should use isMultipleOrTags here.

Comment (jzhangs) [reply src/Select.jsx]: There's some problem about tags here. As a user, when I use 
* 'multiple' Select:  input only used as filter. It's good and not that annoying to always keep filter there.
* 'tags' Select, inputs can be transformed to a selected item. If just update 'isMultiple' to 'isMultipleOrTags', after a random input and select, input was kept there to filter out only the added item. It's definitely not what users want.

Comment (jzhangs) [reply src/Select.jsx]: So we may
* update 'multiple' for now
* add some prop/config as suggested in https://github.com/ant-design/ant-design/issues/9573
* or make tags smarter - keep filter only when the input value is in previous select options

What do you think?

Comment (yutingzhao1991) [reply src/Select.jsx]: I think add a props `keepSearchText` is enough. What's your option @afc163 @yesmeck

Comment (yesmeck) [reply src/Select.jsx]: Yes, we need to introduce a new prop to enable this behavior.

Comment (jzhangs) [reply src/Select.jsx]: Maybe `keepInputFilter: 'always' | 'no' | 'smart'` ? I prefer the `smart keep/clear search input` way on `tags` though it's not that easy to implement.  And, as there're already lots of Select props, still tend not to add more.

Leave further work to you @yutingzhao1991 .

Comment (yutingzhao1991) [reply src/Select.jsx]: I think `keepSearchInput: true | false` is enough.

Comment (afc163) [reply src/Select.jsx]: react-component/select@7531b4d

Add a prop of `autoClearSearchValue`, default `false`.

**Pull request merged**